src/Microsoft.VisualStudio.Editors/Common/Utils.vb

```diff
@@ -1665,5 +1666,29 @@ Namespace Microsoft.VisualStudio.Editors.Common
 
         End Class
 #End Region
+
+        Public Class SerializationProvider
```

Comment:
Do you think unit tests would add value here? For example, to validate that the types we know about round trip successfully.

Reply:
Yeah. I want to do that. But I also wanted to get this PR available to review. So, I'll be adding tests shortly.

Reply:
Unit tests have been added. I did a few basic types, a couple complex types, and null/empty stream testing.